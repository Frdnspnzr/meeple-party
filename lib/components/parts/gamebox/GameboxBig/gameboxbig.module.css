.card {
  width: 600px;
  height: 252px;
  display: grid;
  background-color: var(--bs-gray-100);
  grid-template-columns: 250px 60px 1fr 1fr 1fr;
  grid-template-rows: 4rem 1fr 1fr;
  grid-template-areas:
    "image info title title title"
    "image info status-own status-wanttoplay status-wishlist"
    "image info friends-own friends-wanttoplay friends-wishlist";

  @media screen and (max-width: 768px) {
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 250px 60px 1fr 1fr 1fr 1fr;
    grid-template-areas:
      "image image"
      "title title"
      "info info"
      "status-own friends-own"
      "status-wanttoplay friends-wanttoplay"
      "status-wishlist friends-wishlist";

    gap: 0.25rem;
    width: 95vw;
    height: unset;
  }
}

.image {
  width: 250px;
  height: 250px;
  object-fit: cover;
  @media screen and (max-width: 768px) {
    width: 100%;
    height: 100%;
  }
}

.imagePosition {
  grid-area: image;
}

.name {
  grid-area: title;
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-height: 3rem;
  overflow: hidden;
  font-weight: bold;
  margin: 0;
  padding: 0;
  justify-self: start;
  align-self: center;
  padding-left: 0.5rem;
  padding-right: 0.5rem;

  @media screen and (max-width: 768px) {
    justify-self: center;
  }
}

.name a {
  color: var(--bs-text);
  text-decoration: none;
}

.info {
  grid-area: info;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  border-right: 1px solid var(--bs-border-color);

  @media screen and (max-width: 768px) {
    flex-direction: row;
  }
}

.metricDivider {
  width: 80%;
  margin: 0;

  @media screen and (max-width: 768px) {
    display: none;
  }
}

.metricTitle {
  font-size: 1.25rem;
  text-align: center;
  font-weight: normal;
}

.metricLabel {
  font-size: 0.6rem;
  text-align: center;
  color: var(--bs-light-text);
}

.myStatusOwn,
.myStatusWantToPlay,
.myStatusWishlist {
  border-top: 1px solid var(--bs-border-color);

  @media screen and (max-width: 768px) {
    border: 0;
  }
}

.myStatusOwn {
  grid-area: status-own;
}

.myStatusWantToPlay {
  grid-area: status-wanttoplay;
}

.myStatusWishlist {
  grid-area: status-wishlist;
}

.friendStatus {
  display: flex;
  flex-direction: column;
  justify-content: start;
  align-items: center;
  gap: 1rem;
  border: transparent;
  background: transparent;
}

.friendStatusOwn {
  grid-area: friends-own;
}

.friendStatusWantToPlay {
  grid-area: friends-wanttoplay;
}

.friendStatusWishlist {
  grid-area: friends-wishlist;
}

.statusButton {
  font-size: 1.75rem;
  place-self: center stretch;
}

.statusText {
  font-size: 0.7rem;
  text-align: center;
  font-weight: lighter;
}

.statusFriends {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  grid-area: "friends";
}
