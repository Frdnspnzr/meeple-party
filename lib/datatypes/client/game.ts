export interface Game {
  id: number;
  name: string;
  thumbnail?: string;
  image?: string;
  year: number;
  playingTime: number;
  minPlayers: number;
  maxPlayers: number;
  weight: number;
  BGGRating?: number;
  BGGRank?: number;
}
