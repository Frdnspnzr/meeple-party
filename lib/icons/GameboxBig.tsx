export default function IconGameBoxBig() {
  return <i className="bi bi-grid-fill"></i>;
}
