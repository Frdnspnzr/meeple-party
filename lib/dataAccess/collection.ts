import { GameCollectionResult } from "@/app/api/v2/user/[userId]/collection/route";
import axios from "axios";
import { UserId } from "../datatypes/client/userProfile";

export default function getCollection(
  userId: UserId
): Promise<GameCollectionResult> {
  return axios
    .get<GameCollectionResult>(`/api/v2/user/${userId}/collection`)
    .then((response) => response.data);
}
