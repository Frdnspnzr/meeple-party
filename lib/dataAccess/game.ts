import { GameGetResult } from "@/app/api/v2/game/[gameId]/route";
import { Game as ServerGame } from "@prisma/client";
import axios from "axios";
import { Game } from "../datatypes/client/game";

export function getGame(gameId: number): Promise<Game> {
  return axios
    .get<GameGetResult>(`/api/v2/game/${gameId}`)
    .then((response) => response.data.game)
    .then(convertGame);
}

export function convertGame(serverGame: ServerGame): Game {
  return {
    id: serverGame.id,
    maxPlayers: serverGame.maxPlayers,
    minPlayers: serverGame.minPlayers,
    name: serverGame.name,
    playingTime: serverGame.playingTime,
    weight: serverGame.weight,
    year: serverGame.year,
    BGGRank: serverGame.BGGRank || undefined,
    BGGRating: serverGame.BGGRating || undefined,
    image: serverGame.image || undefined,
    thumbnail: serverGame.thumbnail || undefined,
  };
}
