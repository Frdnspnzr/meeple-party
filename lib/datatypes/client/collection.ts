import { GameId } from "./game";
import { UserId } from "./userProfile";

export interface Collection {
  user: UserId;
  own: GameId[];
  wantToPlay: GameId[];
  wishlist: GameId[];
}