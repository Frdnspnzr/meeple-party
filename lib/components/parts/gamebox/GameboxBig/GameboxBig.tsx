"use client";

import AvatarStack from "@/components/AvatarStack/AvatarStack";
import PersonList from "@/components/PersonList/PersonList";
import { useModal } from "@/context/modalContext";
import { GameCollectionStatus, StatusByUser } from "@/datatypes/collection";
import { Game } from "@/datatypes/game";
import { UserProfile } from "@/datatypes/userProfile";
import { useTranslation } from "@/i18n/client";
import classNames from "classnames";
import Image from "next/image";
import Link from "next/link";
import { useCallback, useMemo } from "react";
import StatusButton from "../../StatusButton/StatusButton";
import GameboxProps from "../interface";
import styles from "./gameboxbig.module.css";

export default function GameboxBig({
  game,
  friendCollections,
  myCollection,
  updateStatus,
}: Readonly<GameboxProps>) {
  return (
    <div className={classNames("card", styles.card)}>
      {game.image ? (
        <Link href={`/app/game/${game.id}`} className={styles.imagePosition}>
          <Image
            src={game.image}
            alt={game.name}
            width={250}
            height={250}
            unoptimized
            className={classNames(styles.image, "rounded-start")}
          />
        </Link>
      ) : (
        <div className={classNames(styles.image, "rounded-start")} />
      )}
      <MetricList game={game} />

      <h2 className={styles.name}>
        <Link href={`/app/game/${game.id}`}>{game.name}</Link>
      </h2>

      <StatusList
        game={game}
        friendCollection={friendCollections}
        myCollection={myCollection}
        updateStatus={updateStatus}
      />
    </div>
  );
}

function MetricList({ game }: Readonly<{ game: Game }>) {
  const { t } = useTranslation("game");
  return (
    <div className={styles.info}>
      <Metric
        text={round(game.weight).toString()}
        label={t("Attributes.Weight")}
      />
      <hr className={styles.metricDivider} />
      <Metric
        text={game.playingTime.toString()}
        label={t("Attributes.PlayingTime")}
      />
      <hr className={styles.metricDivider} />
      <Metric
        text={
          game.minPlayers === game.maxPlayers
            ? game.minPlayers.toString()
            : `${game.minPlayers}-${game.maxPlayers}`
        }
        label={t("Attributes.Players")}
      />
    </div>
  );
}

function Metric({ text, label }: Readonly<{ text: string; label: string }>) {
  return (
    <div className={styles.metric}>
      <div className={styles.metricTitle}>{text}</div>
      <div className={styles.metricLabel}>{label}</div>
    </div>
  );
}

function StatusList({
  game,
  friendCollection,
  myCollection,
  updateStatus,
}: Readonly<{
  game: Game;
  friendCollection?: StatusByUser;
  myCollection: GameCollectionStatus;
  updateStatus?: (status: Partial<GameCollectionStatus>) => void;
}>) {
  return (
    <>
      <Status
        game={game}
        status="own"
        friends={friendCollection ? friendCollection.own : []}
        myCollection={myCollection}
        updateStatus={updateStatus}
      />
      <Status
        game={game}
        status="wanttoplay"
        friends={friendCollection ? friendCollection.wantToPlay : []}
        myCollection={myCollection}
        updateStatus={updateStatus}
      />
      <Status
        game={game}
        status="wishlist"
        friends={friendCollection ? friendCollection.wishlist : []}
        myCollection={myCollection}
        updateStatus={updateStatus}
      />
    </>
  );
}

function Status({
  game,
  friends,
  status,
  myCollection,
  updateStatus,
}: Readonly<{
  game: Game;
  friends: UserProfile[];
  myCollection: GameCollectionStatus;
  status: "own" | "wishlist" | "wanttoplay";
  updateStatus?: (status: Partial<GameCollectionStatus>) => void;
}>) {
  const { t } = useTranslation("default");
  const { t: ct } = useTranslation("collection");
  const { open: openModal, close: closeModal } = useModal();
  const state = useMemo(() => {
    if (status === "own") {
      return myCollection.own;
    } else if (status === "wanttoplay") {
      return myCollection.wantToPlay;
    } else if (status === "wishlist") {
      return myCollection.wishlist;
    }
  }, [myCollection, status]);

  const translationBaseKey = useMemo(() => {
    if (status === "own") {
      return "Own";
    } else if (status === "wanttoplay") {
      return "WantToPlay";
    } else if (status === "wishlist") {
      return "Wishlist";
    }
  }, [status]);

  const label = useMemo(() => {
    if (status === "own") {
      return ct("FriendCollections.Own");
    } else if (status === "wanttoplay") {
      return ct("FriendCollections.WantToPlay");
    } else if (status === "wishlist") {
      return ct("FriendCollections.Wishlist");
    }
  }, [status, ct]);

  const toggle = useCallback(() => {
    if (!updateStatus) return;
    if (status === "own") {
      updateStatus({
        own: !myCollection.own,
        wantToPlay: myCollection.wantToPlay,
        wishlist: myCollection.wishlist,
      });
    } else if (status === "wanttoplay") {
      updateStatus({
        own: myCollection.own,
        wantToPlay: !myCollection.wantToPlay,
        wishlist: myCollection.wishlist,
      });
    } else if (status === "wishlist") {
      updateStatus({
        own: myCollection.own,
        wantToPlay: myCollection.wantToPlay,
        wishlist: !myCollection.wishlist,
      });
    }
  }, [myCollection, status, updateStatus]);

  const showFriends = useCallback(() => {
    openModal({
      title: ct(`FriendCollections.${translationBaseKey}`, {
        game: game.name,
      }),
      content: <PersonList persons={friends} onClick={closeModal} />,
    });
  }, [friends, game.name, ct, openModal, closeModal, translationBaseKey]);

  return (
    <>
      <div
        className={classNames({
          [styles.myStatusOwn]: status === "own",
          [styles.myStatusWantToPlay]: status === "wanttoplay",
          [styles.myStatusWishlist]: status === "wishlist",
        })}
      >
        <StatusButton
          status={status}
          className={styles.statusButton}
          active={state || false}
          toggle={toggle}
        />
        <div className={styles.statusText}>
          {state
            ? t(`States.${translationBaseKey}`)
            : t(`States.Not${translationBaseKey}`)}
        </div>
      </div>
      {friends.length > 0 && (
        <button
          className={classNames(styles.friendStatus, {
            [styles.friendStatusOwn]: status === "own",
            [styles.friendStatusWantToPlay]: status === "wanttoplay",
            [styles.friendStatusWishlist]: status === "wishlist",
          })}
          onClick={showFriends}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              showFriends();
            }
          }}
          tabIndex={0}
          data-testid={`friends-${translationBaseKey}`}
          aria-label={label}
        >
          <AvatarStack
            max={4}
            distance="sm"
            avatars={friends.map((f) => ({
              id: f.id,
              name: f.name || "",
              image: f.image,
            }))}
          />
          <div className={styles.statusText}>
            {t(`FriendStates.${translationBaseKey}`, { count: friends.length })}
          </div>
        </button>
      )}
    </>
  );
}

function round(x: number): number {
  return Math.round(x * 10) / 10;
}
