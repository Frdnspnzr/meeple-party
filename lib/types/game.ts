import * as client from "@prisma/client";

/**
 * General game information known to Meeple Party
 */
export type Game = Omit<client.Game, "updatedAt">;

/**
 * Game information as found on BoardGameGeek
 */
export interface BggGame extends Omit<Game, "wikidataId"> {
  description: string;
  designers: string[];
  artists: string[];
  age: number;
}

/**
 * Game information with additional data
 */
export type ExpandedGame = Game & {
  names: Omit<client.AlternateGameName, "gameId">[];
};

/**
 * Unique ID of a game. Equal to that game's ID on BoardGameGeek.
 */
export type GameId = number;

/**
 * Game data as written to the database.
 */
export type PrismaGame = client.Game;

/**
 * Game data as wirtten to the database, including alternate names.
 */
export type PrismaGameWithNames = PrismaGame & {
  alternateNames: client.AlternateGameName[];
};

/**
 * Converts a {@link PrismaGame} to {@link Game}, dropping all superfluous attributes.
 * @param game Game in database format to convert.
 */
export function prismaGameToGame(game: client.Game): Game {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
  };
}

/**
 * Converts a {@link PrismaGameWithNames} to {@link ExpandedGame}, renaming attirbutes and dropping all superfluous attributes.
 * @param game Game in database format to convert.
 */
export function prismaGameToExpandedGame(
  game: PrismaGameWithNames
): ExpandedGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
    names: game.alternateNames.map((n) => ({
      language: n.language,
      name: n.name,
    })),
  };
}

/**
 * Converts a {@link BggGame} to {@link ExpandedGame}, dropping all superfluous attributes.
 * @param game Game from BoardGameGeek to convert.
 */
export function bggGameToExpandedGame(game: BggGame): ExpandedGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: null,
    year: game.year,
    names: [],
  };
}

/**
 * Converts a {@link ExpandedGame} to {@link Game}, dropping all superfluous attributes
 * @param game Game to convert.
 */
export function expandedGameToGame(game: ExpandedGame): Game {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    wikidataId: game.wikidataId,
    year: game.year,
  };
}

/**
 * Converts a {@link Game} to {@link ExpandedGame}, initializing all needed attributes empty or undefined.
 * @param game Game to convert.
 */
export function gameToExpandedGame(game: Game): ExpandedGame {
  return { ...game, names: [] };
}

/**
 * Converts a {@link ExpandedGame} to {@link BggGame}, dropping all superfluous attributes and initializing all additional attributes empty or undefined.
 * @param game Game to convert
 */
export function expandedGameToBggGame(game: ExpandedGame): BggGame {
  return {
    BGGRank: game.BGGRank,
    BGGRating: game.BGGRating,
    image: game.image,
    id: game.id,
    maxPlayers: game.maxPlayers,
    minPlayers: game.minPlayers,
    name: game.name,
    playingTime: game.playingTime,
    thumbnail: game.thumbnail,
    weight: game.weight,
    year: game.year,
    age: 0,
    artists: [],
    description: "",
    designers: [],
  };
}
