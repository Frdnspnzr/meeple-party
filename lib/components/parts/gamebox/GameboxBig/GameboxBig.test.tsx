import Avatar from "@/components/Avatar/Avatar";
import { ModalConfiguration, useModal } from "@/context/modalContext";
import { GameCollectionStatus, StatusByUser } from "@/datatypes/collection";
import { Game } from "@/datatypes/game";
import {
  generateArray,
  generateBoolean,
  generateUserProfile,
  render,
} from "@/utility/test";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import GameboxBig from "./GameboxBig";

jest.mock("@/i18n/client");
jest.mock("@/hooks/api/useCollectionStatus");
jest.mock("@/components/Avatar/Avatar", () => ({
  __esModule: true,
  namedExport: jest.fn(),
  default: jest.fn(),
}));
jest.mock("@/context/modalContext");

const game: Game = {
  id: 123,
  name: "Test game",
  BGGRank: 23,
  BGGRating: 8.4,
  minPlayers: 1,
  maxPlayers: 5,
  playingTime: 120,
  weight: 3.8,
  year: 2018,
  image: "testimage",
  thumbnail: "testthumbnail",
};
const friendCollections: StatusByUser = {
  own: generateArray(generateUserProfile),
  wantToPlay: generateArray(generateUserProfile),
  wishlist: generateArray(generateUserProfile),
};
const myCollection: GameCollectionStatus = {
  own: true,
  wantToPlay: false,
  wishlist: true,
};

describe("GameBoxBig", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
    jest.mocked(Avatar).mockImplementation(() => <>Avatar</>);
  });
  beforeEach(() => {
    jest.mocked(useModal).mockImplementation(() => ({
      open: () => {},
      close: () => {},
    }));
  });
  afterEach(() => {
    jest.resetAllMocks();
  });
  it("Matches snapshot", async () => {
    const { container: rendered } = render(
      <GameboxBig
        game={game}
        friendCollections={friendCollections}
        myCollection={myCollection}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it("Matches snapshot for precise weight", async () => {
    const { container: rendered } = render(
      <GameboxBig
        game={{ ...game, weight: Math.PI }}
        friendCollections={friendCollections}
        myCollection={myCollection}
      />
    );
    expect(rendered).toMatchSnapshot("default game");
  });
  test.each(["own", "wanttoplay", "wishlist"])(
    "opens modal when clicking friends list for %s status",
    async () => {
      const user = userEvent.setup();
      const translationKey =
        status === "own"
          ? "Own"
          : status === "wanttoplay"
            ? "WantToPlay"
            : "Wishlist";
      const expectedFriends =
        status === "own"
          ? friendCollections.own
          : status === "wanttoplay"
            ? friendCollections.wantToPlay
            : friendCollections.wishlist;
      let actualConfiguration: ModalConfiguration | null = null;
      const useModalMock = jest.mocked(useModal);
      useModalMock.mockImplementation(() => ({
        open: (configuration) => {
          actualConfiguration = configuration;
        },
        close: () => {},
      }));
      render(
        <GameboxBig
          game={game}
          friendCollections={friendCollections}
          myCollection={myCollection}
        />
      );
      const button = screen.getByRole("button", {
        name: `FriendCollections.${translationKey}`,
      });
      await user.click(button);

      expect(useModalMock).toHaveBeenCalledTimes(3);
      expect(actualConfiguration).not.toBeNull();
      expect(actualConfiguration!.title).toBe(
        `FriendCollections.${translationKey}`
      );
      render(actualConfiguration!.content);
      expectedFriends.forEach((friend) => {
        const link = screen.getByText(friend.name!);
        expect(link).not.toBeNull();
      });
    }
  );
  test.each(["own", "wanttoplay", "wishlist"])(
    "calls update callback when status button for %s is clicked",
    async (status) => {
      const user = userEvent.setup();
      let actualStatus = null;
      const callback = (status: Partial<GameCollectionStatus>) => {
        actualStatus = status;
      };
      const randomCollection = {
        own: generateBoolean(),
        wantToPlay: generateBoolean(),
        wishlist: generateBoolean(),
      };

      render(
        <GameboxBig
          game={game}
          friendCollections={friendCollections}
          myCollection={randomCollection}
          updateStatus={callback}
        />
      );
      const button = screen.getByRole("button", {
        name:
          status === "own"
            ? "States.Own"
            : status === "wanttoplay"
              ? "States.WantToPlay"
              : "States.Wishlist",
      });
      await user.click(button);

      expect(actualStatus).not.toBeNull();

      if (status === "own") {
        expect(actualStatus!.own).toBe(!randomCollection.own);
      } else if (status === "wanttoplay") {
        expect(actualStatus!.wantToPlay).toBe(!randomCollection.wantToPlay);
      } else if (status === "wishlist") {
        expect(actualStatus!.wishlist).toBe(!randomCollection.wishlist);
      }
    }
  );
});
