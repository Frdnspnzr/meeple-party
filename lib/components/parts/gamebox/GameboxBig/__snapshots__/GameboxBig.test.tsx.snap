// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoxBig Matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="card card"
  >
    <a
      class="imagePosition"
      href="/app/game/123"
    >
      <img
        alt="Test game"
        class="image rounded-start"
        data-nimg="1"
        decoding="async"
        height="250"
        loading="lazy"
        src="testimage"
        style="color: transparent;"
        width="250"
      />
    </a>
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.8
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      <a
        href="/app/game/123"
      >
        Test game
      </a>
    </h2>
    <div
      class="myStatusOwn"
    >
      <button
        aria-label="States.Own"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.Own
      </div>
    </div>
    <button
      aria-label="FriendCollections.Own"
      class="friendStatus friendStatusOwn"
      data-testid="friends-Own"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="statusText"
      >
        FriendStates.Own
      </div>
    </button>
    <div
      class="myStatusWantToPlay"
    >
      <button
        aria-label="States.WantToPlay"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.NotWantToPlay
      </div>
    </div>
    <button
      aria-label="FriendCollections.WantToPlay"
      class="friendStatus friendStatusWantToPlay"
      data-testid="friends-WantToPlay"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="statusText"
      >
        FriendStates.WantToPlay
      </div>
    </button>
    <div
      class="myStatusWishlist"
    >
      <button
        aria-label="States.Wishlist"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.Wishlist
      </div>
    </div>
    <button
      aria-label="FriendCollections.Wishlist"
      class="friendStatus friendStatusWishlist"
      data-testid="friends-Wishlist"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        Avatar
        Avatar
        Avatar
        Avatar
      </div>
      <div
        class="statusText"
      >
        FriendStates.Wishlist
      </div>
    </button>
  </div>
</div>
`;

exports[`GameBoxBig Matches snapshot for precise weight: default game 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="card card"
  >
    <a
      class="imagePosition"
      href="/app/game/123"
    >
      <img
        alt="Test game"
        class="image rounded-start"
        data-nimg="1"
        decoding="async"
        height="250"
        loading="lazy"
        src="testimage"
        style="color: transparent;"
        width="250"
      />
    </a>
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.1
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      <a
        href="/app/game/123"
      >
        Test game
      </a>
    </h2>
    <div
      class="myStatusOwn"
    >
      <button
        aria-label="States.Own"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.Own
      </div>
    </div>
    <button
      aria-label="FriendCollections.Own"
      class="friendStatus friendStatusOwn"
      data-testid="friends-Own"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Own
      </div>
    </button>
    <div
      class="myStatusWantToPlay"
    >
      <button
        aria-label="States.WantToPlay"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.NotWantToPlay
      </div>
    </div>
    <button
      aria-label="FriendCollections.WantToPlay"
      class="friendStatus friendStatusWantToPlay"
      data-testid="friends-WantToPlay"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.WantToPlay
      </div>
    </button>
    <div
      class="myStatusWishlist"
    >
      <button
        aria-label="States.Wishlist"
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </button>
      <div
        class="statusText"
      >
        States.Wishlist
      </div>
    </div>
    <button
      aria-label="FriendCollections.Wishlist"
      class="friendStatus friendStatusWishlist"
      data-testid="friends-Wishlist"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Wishlist
      </div>
    </button>
  </div>
</div>
`;
